add tasks, validators, errors

AlmazYarullin opened the pull request.

AlmazYarullin committed:
add tasks, validators, errors

DmitryBarskov commented on app/controllers/tasks_controller.rb:
```diff
@@ -0,0 +1,47 @@
+  end
+
+  def update
+    if @task.update(task_params)
+      redirect_to task_path(@task), notice: 'Task has been updated'
+    end
```
What happens if task is failed to update?

AlmazYarullin replied:
Ooops, it should render edit page. I'll fix it

DmitryBarskov commented on app/views/tasks/show.html.erb:
```diff
@@ -0,0 +1,24 @@
+  <%= @task.deadline_at %>
+</p>
+
+<%= link_to 'Edit', edit_task_path(@task) %>
+<%= link_to 'Back', tasks_path %>
```
It would be convenient to see destroy action here.

AlmazYarullin replied:
It will be done

AlmazYarullin committed:
add features, fix

Feature: New button Destroy when showing task.
Fixed problem: if task cant be updated in edit page, user will remain in edit page
